selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/Workspaces.java

```diff
@@ -300,4 +367,84 @@ public void clickOnDeleteButtonInDialogWindow() {
 
     return titles;
   }
+
+  public int getVisibleWorkspacesCount() {
+    return seleniumWebDriverHelper
+        .waitVisibilityOfAllElements(
+            By.xpath("//div[@class='workspace-name-clip']/parent::div/parent::div/parent::div"))
+        .size();
+  }
+
+  public String getFullNameOfWorkspacesListItem(int index) {
+    String itemXpath = String.format(Locators.WORKSPACE_LIST_ITEM, index);
+    String fullNameXpath = itemXpath + "//div[@class='workspace-name-clip']";
+    return seleniumWebDriverHelper
+        .waitVisibility(By.xpath(fullNameXpath))
```

Comment:
It seems can be simplified by using `seleniumWebDriverHelper.waitVisibilityAndGetAttribute()` method